data_diff/database.py

```diff
@@ -261,11 +380,45 @@ def quote(self, s: str):
     def to_string(self, s: str):
         return f"cast({s} as varchar(1024))"
 
+    def select_table_schema(self, path: DbPath) -> str:
+        if len(path) > 1:
+            raise ValueError("Unexpected table path for oracle")
+        (table,) = path
+
+        return (
+            f"SELECT column_name, data_type, 9 as datetime_precision, data_precision as numeric_precision"
+            f" FROM USER_TAB_COLUMNS WHERE table_name = '{table.upper()}'"
+        )
+
+    def normalize_value_by_type(self, value, coltype: ColType) -> str:
+        if isinstance(coltype, PrecisionType):
+            return f"to_char(cast({value} as timestamp), 'YYYY-MM-DD HH24:MI:SS.FF{coltype.precision or ''}')"
+        return self.to_string(f"{value}")
+
+    def _parse_type(self, type_repr: str, datetime_precision: int = None, numeric_precision: int = None) -> ColType:
+        """ """
+        regexps = {
+            r"TIMESTAMP\((\d)\) WITH LOCAL TIME ZONE": Timestamp,
+            r"TIMESTAMP\((\d)\) WITH TIME ZONE": TimestampTZ,
+        }
+        for regexp, cls in regexps.items():
+            m = re.match(regexp + "$", type_repr)
+            if m:
+                datetime_precision = int(m.group(1))
+                return cls(precision=datetime_precision or DEFAULT_PRECISION)
+
+        return UnknownColType(type_repr)
+
 
 class Redshift(Postgres):
     def md5_to_int(self, s: str) -> str:
         return f"strtol(substring(md5({s}), {1+MD5_HEXDIGITS-CHECKSUM_HEXDIGITS}), 16)::decimal(38)"
 
+    def normalize_value_by_type(self, value, coltype: ColType) -> str:
+        if coltype.precision != 6:
+            raise TypeError(f"Bad precision for Redshift: {coltype})")
```

Comment:
👍🏻  Consider adding a comment here with the bug that's preventing this

Reply:
Technically we can also support precision = 3. But I don't know if it's worth the effort.